### ISSUE TITLE
Reused connection resend the old CURLOPT_XOAUTH2_BEARER token instead of new ones.

### ISSUE THREAD

**Issue by yushizhao** [curl/curl#7722]
It looks like https://github.com/curl/curl/issues/1685.

Libcurl will resend the token set in the first request even if in futher requests, new tokens are explicitly curl_easy_setopt with.

Everything works fine if I set CURLOPT_FORBID_REUSE. But if I do want to reuse the connection, is there a workaground? I am using version 7.68.0.

**Comment by bagder** [comment]
Thanks a lot for your report!

Can you to provide us with more details on how to go ahead and repeat this. Preferably with a command line or a stand-alone program we can run from our ends against a public URL to trigger the problem?

**Comment by yushizhao** [comment]
```c
#include <curl/curl.h>

int main(int argc, char *argv[])
{
  CURLcode ret;
  CURL *curl;

  curl_global_init(CURL_GLOBAL_ALL);

#define URL "http://httpbin.org/"

  curl = curl_easy_init();
  curl_easy_setopt(curl, CURLOPT_VERBOSE, 1L);
  curl_easy_setopt(curl, CURLOPT_URL, URL);
  curl_easy_setopt(curl, CURLOPT_USERAGENT, "curl/7.68.0");
  curl_easy_setopt(curl, CURLOPT_XOAUTH2_BEARER, "token AAAAAAAAAAAA");
  curl_easy_setopt(curl, CURLOPT_HTTPAUTH, CURLAUTH_BEARER);
  curl_easy_perform(curl);

  curl_easy_reset(curl);

  curl_easy_setopt(curl, CURLOPT_VERBOSE, 1L);
  curl_easy_setopt(curl, CURLOPT_URL, URL);
  curl_easy_setopt(curl, CURLOPT_USERAGENT, "curl/7.68.0");
  curl_easy_setopt(curl, CURLOPT_XOAUTH2_BEARER, "token BBBBBBBBBBBB");
  curl_easy_setopt(curl, CURLOPT_HTTPAUTH, CURLAUTH_BEARER);
  curl_easy_perform(curl);

  curl_easy_cleanup(curl);

  curl_global_cleanup();

  return 0;
}
```
Should be able to see `token AAAAAAA...` has been sent twice.

**Comment by bagder** [comment]
I cannot reproduce this with latest libcurl. I suspect your issue might have been fixed since 7.68.0.

**Comment by yushizhao** [comment]
Yes. I tried 7.79.0. Works fine. Please close this issue. Thank you!
